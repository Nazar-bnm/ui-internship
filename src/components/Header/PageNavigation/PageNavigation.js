import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const PageNavigation = (props) => {
  return (
    <ul className={'list container'}>
      {props.options.map(({ value, label, link }, i) => (
        <li key={i.toString()}
          value={value}
          className={'list__item'}
        >
          <Link to={link}>{label}</Link>
        </li>
      ))}
    </ul>
  );
};

PageNavigation.propTypes = {
  class: PropTypes.string,
  options: PropTypes.array,
};

export default PageNavigation;
