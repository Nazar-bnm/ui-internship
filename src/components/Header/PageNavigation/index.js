import React from 'react';
import PropTypes from 'prop-types';

const PageNavigation = (props) => {
  return (
    <ul className={props.class}>
      {props.options.map((option, i) => (
        <li key={i.toString()}
          value={option.value}
        >
          {option.label}
        </li>
      ))}
    </ul>
  );
};

PageNavigation.propTypes = {
  class: PropTypes.string,
  options: PropTypes.array,
};

export default PageNavigation;
