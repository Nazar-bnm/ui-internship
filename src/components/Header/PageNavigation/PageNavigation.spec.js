import React from 'react';
import { shallow } from 'enzyme';
import PageNavigation from './PageNavigation';

describe('<PageNavigation />', () => {
  let props;
  let wrapper;

  beforeEach(() => {
    props = {
      options: [
        { value: 'women', label: 'women', link: '/404' },
        { value: 'men', label: 'men', link: '/404' },
        { value: 'kids', label: 'kids', link: '/404' },
        { value: 'accs', label: 'accessories', link: '/404' },
        { value: 'new', label: 'what\'s new', link: '/404' },
        { value: 'brands', label: 'brands', link: '/404' },
        { value: 'sale', label: 'sale', link: '/404' },
        { value: 'blog', label: 'blog', link: '/404' },
      ],
    };

    wrapper = shallow(
      <PageNavigation {...props} />
    );
  });
  test('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
