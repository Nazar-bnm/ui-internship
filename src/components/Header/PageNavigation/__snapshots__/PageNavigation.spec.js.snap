// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageNavigation /> should match snapshot 1`] = `
<ul
  className="list container"
>
  <li
    className="list-item"
    key="0"
    value="women"
  >
    <Link
      to="/404"
    >
      women
    </Link>
  </li>
  <li
    className="list-item"
    key="1"
    value="men"
  >
    <Link
      to="/404"
    >
      men
    </Link>
  </li>
  <li
    className="list-item"
    key="2"
    value="kids"
  >
    <Link
      to="/404"
    >
      kids
    </Link>
  </li>
  <li
    className="list-item"
    key="3"
    value="accs"
  >
    <Link
      to="/404"
    >
      accessories
    </Link>
  </li>
  <li
    className="list-item"
    key="4"
    value="new"
  >
    <Link
      to="/404"
    >
      what's new
    </Link>
  </li>
  <li
    className="list-item"
    key="5"
    value="brands"
  >
    <Link
      to="/404"
    >
      brands
    </Link>
  </li>
  <li
    className="list-item"
    key="6"
    value="sale"
  >
    <Link
      to="/404"
    >
      sale
    </Link>
  </li>
  <li
    className="list-item"
    key="7"
    value="blog"
  >
    <Link
      to="/404"
    >
      blog
    </Link>
  </li>
</ul>
`;
